# Black Hole Thermodynamics

## Black Hole Entropy

A black hole is a region of space where gravity is so strong that nothing, not
even light, can escape. Throwing something into a black hole is therefore an
irreversible process, at least in the everyday sense of the word. In fact, it
is irreversible in the thermodynamics sense as well: Adding mass to a black
hole increases the black hole's entropy. It turns out that there's no way to
tell (from the outside) what kind of matter has gone into making a black hole
(charge and angular momentum of matter that enters the black hole is
conserved, but let's assume both are zero). Therefore, the entropy of a black
hole must be greater than the entropy of any conceivable type of matter that
could have been used to create it! Knowing this, it's not hard to estimate the
entropy of a black hole.

### (A)
Use dimensional analysis to show that a black hole of mass $M$ should have a
radius on the order of $\frac{GM}{c^2}$, where $G$ is Newton's gravitational
constant and $c$ is the speed of light. Calculate the approximate radius of a
one-solar-mass black hole

$$
\left.\begin{aligned}\
  [G] &= \text{m}^{3} \cdot \text{kg}^{-1} \cdot \text{s}^{-2} \\
  [M] &= \text{kg} \\
  [c] &= \text{m} \cdot \text{s}^{-1}
\end{aligned}\right\}\Rightarrow
\left[\frac{GM}{c^2}\right]
= \frac{\text{m}^{3}\cdot\text{s}^{-2}}{\text{m}^{2} \cdot \text{s}^{-2}}
= \text{m}
$$

$$
\begin{aligned}
  K = \frac{1}{2}mv^2 &= \frac{GMm}{r_\text{s}} = U \\
  \frac{1}{2}c^2 &= \frac{GM}{r_\text{s}} \\
\end{aligned}
$$

$$
  \boxed{r_\text{s} = \frac{2GM}{c^2}}
$$

In [ ]:
from astropy.constants import G, M_sun, c
from IPython.display import Markdown, display

def schwarzschild(mass):
    return 2*G*mass/c**2

def sci_tex(x, units=str(), digits=4):
    mantissa, exponent = f'{x:.{digits-1}e}'.split('e')
    return rf'{mantissa}\times 10^{{{int(exponent)}}}\,\mathrm{{{units}}}'

r_s = schwarzschild(M_sun).value
G_tex = sci_tex(G.value, units = r'm^3 \cdot kg^{-1} \cdot s^{-2}')
M_tex = r'M_\odot &\approx ' + sci_tex(M_sun.value, r'kg')
c_tex, r_tex = sci_tex(c.value, units = r'm \cdot s^{-1}'), sci_tex(r_s, r'm')

display(Markdown(
  r'''$$
  \left.\begin{aligned}
      G &\approx %s \\ M = %s \\ c &\approx %s
  \end{aligned}\right\}\Rightarrow\left\{\begin{aligned}
      r_s = \frac{2GM}{c^2} &= \frac{2G M_\odot}{c^2} \\ &\approx %s
  \end{aligned}\right. $$ ''' % (G_tex, M_tex, c_tex, r_tex)))

$$
  \left.\begin{aligned}
      G &\approx 6.674\times 10^{-11}\,\mathrm{m^3 \cdot kg^{-1} \cdot s^{-2}} \\ M = M_\odot &\approx 1.988\times 10^{30}\,\mathrm{kg} \\ c &\approx 2.998\times 10^{8}\,\mathrm{m \cdot s^{-1}}
  \end{aligned}\right\}\Rightarrow\left\{\begin{aligned}
      r_s = \frac{2GM}{c^2} &= \frac{2G M_\odot}{c^2} \\ &\approx 2.953\times 10^{3}\,\mathrm{m}
  \end{aligned}\right. $$ 

### (B)
The entropy of an ideal gas or a high temperature Einstein solid is given by
$Nk_\text{B}$ times a logarithm of something. The logarithm is never that large,
so for an order-of-magnitude estimate we can neglect the logarithm and just say
$$ S\sim Nk_\text{B}.$$
Applying this to a black hole, then the entropy of a black hole should be
approximately equal to Boltzmann's constant times the number of particles that
could have gone into making the black hole. For the black hole to be the
maximum entropy possible, we should consider the black hole being made of
particles with the lowest energy: long wavelength photons. But the wavelength
of the photons can't be longer than the size of the black hole. Setting the
total energy equal to $Mc^2$ calculate the approximate number of photons that
could be used to make a black hole of mass $M$. Aside from a factor of $8\pi^2$
your result should agree with the exact formula for the entropy of a black hole:
$\frac{8\pi^2 G M}{h c} k_\text{B}$.

$$
  \lambda \sim r \sim GM c^{-2}
$$

$$
  E_\lambda = hf = \frac{hc}{\lambda} \sim
  \frac{hc^3}{GM}
$$

$$
  NE_\lambda \sim Mc^2
$$

$$
  N \sim \frac{Mc^2}{E_\lambda} \sim \frac{GM^2}{hc}
$$

$$
  S \sim N{k_\text{B}} \sim \frac{GM^2}{hc}k_\text{B}
$$

$$
  S_\text{BH} = \boxed{8\pi^2 \frac{GM^2}{hc}k_\text{B}}
$$

### (C)
Calculate the entropy of a one-solar-mass black hole.

In [10]:
import numpy as np
import astropy.units as u
from astropy.constants import G, M_sun, c, h
from scipy.constants import Boltzmann as k_b
k_b = k_b * u.J / u.K #J/k

#entropy of black hole
def blackHoleEntropy(mass):
  s = 8 * np.pi**2 * G * mass**2 * k_b / (h*c)
  return s

#entropy for solar mass black hole
s_solarmass = blackHoleEntropy(M_sun)
print(s_solarmass)

1.4481509285779017e+54 m2 kg / (K s2)


$$
S = \frac{8\pi^2 G M^2}{hc} k_B
$$

$$
S = \frac{8\pi^2 G M_{\odot}^2}{hc} k_B \approx 1.45 \times 10^{54} \ \text{J/K}
$$



## Black Hole Temperature

### (A)

Use the result of the previous problem to calculate the temperature of a black
hole in terms of its mass $M$ (using $Mc^2$ as its energy).

### (B)

Calculate the temperature of a one-solar-mass black hole.

### (C)

Produce a graph of entropy vs energy and discuss how its shape compares to that
of other systems we have studied this semester.

## Hawking radiation

A black hole is a blackbody if there ever was one, so it should emit blackbody
radiation, called Hawking radiation.  A black hole of mass $M$ has a total
energy of $Mc^2$, a surface area of $\frac{16\pi G^2 M^2}{c^4}$, and a
temperature of $\frac{hc^3}{16\pi^2 k_\text{B}GM}$.

### (A)

Show that the surface area expression really does have units of area.

### (B)

Estimate the typical wavelength of Hawking radiation emitted by a one-solar-mass
black hole. Compare your answer to the size of the black hole.

### (C)

Imagine a black hole in empty space, where it emits radiation but absorbs
nothing. As it loses energy, its mass must decrease: one could say it
"evaporates". Use the intensity of blackbody radiation to derive a differential
equation for the mass as a function of time, and solve this equation to obtain
an expression for the lifetime of a black hole in terms of its initial mass.

### (D)

Calculate the lifetime of a one-solar-mass black hole.